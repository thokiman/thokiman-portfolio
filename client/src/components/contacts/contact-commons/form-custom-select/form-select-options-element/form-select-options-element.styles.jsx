import styled, { keyframes } from "styled-components";

const subColor = "#128194";
const mainColor = "#bccfd5";

export const FormSelectContainer = styled.ul`
  list-style-type: none;
  margin: 5px 0;
  padding: 0;
  width: 100%;
`;

const formSelectItemHover = keyframes`
                0% {
                    transform: rotateX(-100deg);
                    transform-origin: top;
                    opacity: 0;

                }

                100% {
                    transform: rotateX(0deg);
                    transform-origin: top;
                    opacity: 1;

                }

`;
export const FormSelectItem = styled.li`
  cursor: pointer;
  width: 100%;
  animation: ${formSelectItemHover} 1s cubic-bezier(0.175, 0.885, 0.32, 1.275)
    both;
  font-size: 20px;
  color: ${mainColor};
  margin-bottom: 5px;

  &:hover {
    color: ${subColor};
    transition: color linear 0.2s;
  }
`;

FormSelectContainer.displayName = "FormSelectContainer";
FormSelectItem.displayName = "FormSelectItem";
