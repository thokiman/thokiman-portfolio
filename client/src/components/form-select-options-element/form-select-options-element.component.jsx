import React from "react";

import {
  FormSelectContainer,
  FormSelectItem,
} from "./form-select-options-element.styles";

const FormInputSelectOptions = ({ handleClick }) => {
  return (
    <FormSelectContainer>
      <FormSelectItem onClick={handleClick}>Engineering</FormSelectItem>
      <FormSelectItem onClick={handleClick}>
        Web/Mobile Development
      </FormSelectItem>
      <FormSelectItem onClick={handleClick}>Graphic Design</FormSelectItem>
      <FormSelectItem onClick={handleClick}>Photography</FormSelectItem>
    </FormSelectContainer>
  );
};

export default FormInputSelectOptions;
