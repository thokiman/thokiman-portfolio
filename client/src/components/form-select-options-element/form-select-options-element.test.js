import React from "react";
import { shallow } from "enzyme";

import { FormInputSelectOptions } from "./form-select-options-element.component";

describe("FormInputSelectOptions component", () => {
  let wrapper;
  let mockHandleClick;
  beforeEach(() => {
    mockHandleClick = jest.fn();
    const mockProps = {
      handleClick: mockHandleClick,
    };
    wrapper = shallow(<FormInputSelectOptions {...mockProps} />);
  });

  it("should render FormInputSelectOptions component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
