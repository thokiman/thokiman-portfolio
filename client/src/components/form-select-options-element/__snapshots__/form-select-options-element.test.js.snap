// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputSelectOptions component should render FormInputSelectOptions component 1`] = `ShallowWrapper {}`;
